pkg/operator/operator.go

```diff
@@ -133,51 +139,45 @@ func (c *vSphereProblemDetectorController) sync(ctx context.Context, syncCtx fac
 		return err
 	}
 
-	availableCnd := operatorapi.OperatorCondition{
-		Type:   controllerName + operatorapi.OperatorStatusTypeAvailable,
-		Status: operatorapi.ConditionTrue,
+	if !platformSupported {
+		return nil
 	}
 
-	blockUpgrade := false
-	var blockUpgradeReason string
+	clusterInfo := util.NewClusterInfo()
+	delay, checkPerformed := c.runSyncChecks(ctx, clusterInfo)
```

Comment:
`delay` is zero when `runSyncChecks` is run before the next check time. If I am reading the code below right, `time.AfterFunc(delay, ...` will enqueue immediate resync. This resync will enqueue next immediate resync and so on.